pkg/api/datasources.go

```diff
@@ -256,13 +256,11 @@ func (hs *HTTPServer) DeleteDataSourceByName(c *models.ReqContext) response.Resp
 	})
 }
 
-func validateURL(tp string, u string) response.Response {
-	if u != "" {
-		if _, err := datasource.ValidateURL(tp, u); err != nil {
-			datasourcesLogger.Error("Received invalid data source URL as part of data source command",
-				"url", u)
-			return response.Error(400, fmt.Sprintf("Validation error, invalid URL: %q", u), err)
-		}
+func validateURL(cmdType string, url string) response.Response {
+	if _, err := datasource.ValidateURL(cmdType, url); err != nil {
+		datasourcesLogger.Error("Received invalid data source URL as part of data source command",
```

Comment:
Can be discussed if this is needed given that `response.Error(400, fmt.Sprintf("Validation error, invalid URL: %q", url), err)` will log the `err` as error

Reply:
You're right, we're basically logging 3 times the same thing.

![Screen Shot 2022-01-27 at 12 59 05](https://user-images.githubusercontent.com/41638679/151395610-ee4c062a-8e41-42fe-b57e-9ea116c42762.png)